src/django/api/views.py

```diff
@@ -2179,6 +2182,27 @@ def get_link(self, path, method, base_url):
         return None
 
 
+def create_nonstandard_fields(fields, contributor):
+    unique_fields = list(set(fields))
+
+    existing_fields = NonstandardField.objects.filter(
+        contributor=contributor).values_list('column_name', flat=True)
+    new_fields = filter(lambda f: f not in existing_fields,
+                        unique_fields)
```

Comment:
Consider lowercasing all of the field names in the csv before comparing them to existing fields.

I accidentally edited the CSV template, and capitalized the `C` in `country`. After uploading a facility, `Country` now shows up as a non-standard field:

![image](https://user-images.githubusercontent.com/1042475/118698769-caebb600-b7de-11eb-8246-ffe2e33d3134.png)

Reply:
I have added lowercasing the fields when comparing to the standard fields (country, address, etc), to prevent users from accidentally adding one of our standard fields with a casing mistake. 
My thoughts are that we want to leave the casing as submitted for actually saving the column_names, as we use the column_name to access user-submitted data. If the user has a facility with { "column_ABC": "this is data" }, in order to access that data we will need to use facility['column_ABC'] with the same casing. 
I am torn on whether to map over the list of existing fields and lowercase them when comparing, so that having 'column_ABC' will prevent a user from adding 'column_abc'. I am leaning towards not, because the addition of the alternatively cased field to their embed settings will potentially help users see that the field was added incorrectly, especially since they can choose to ignore/hide that field or alter its display field.

Reply:
That approach makes sense to me.